polygon/rest/models/request.py

```diff
@@ -0,0 +1,104 @@
+from typing import Dict, Optional
+
+X_POLYGON_EDGE_ID = "X-Polygon-Edge-ID"
+X_POLYGON_EDGE_IP_ADDRESS = "X-Polygon-Edge-IP-Address"
+X_POLYGON_EDGE_USER_AGENT = "X-Polygon-Edge-User-Agent"
+
+HEADER = "header"
+
+
+class RequestOptionBuilder:
+    def __init__(
+        self,
+        edge_id: Optional[str] = None,
+        edge_ip_address: Optional[str] = None,
+        edge_user: Optional[str] = None,
+    ):
+        """
+        RequestOptionBuilder is a utility class to build polygon api options used in requests.
+        :param edge_id: is a required Launchpad header. It identifies the Edge User requesting data
+        :param edge_ip_address: is a required Launchpad header. It denotes the originating IP Address of the Edge User
+        :param edge_user: is an optional Launchpad header. It denotes the originating UserAgent of the Edge User requesting data.
+        """
+        self.headers: Optional[Dict[str, str]] = None
+        if edge_id is not None and edge_ip_address is not None:
+            self.edge_headers(
+                edge_id=edge_id, edge_ip_address=edge_ip_address, edge_user=edge_user
+            )
+
+    def edge_headers(
+        self,
+        edge_id: Optional[str] = None,
+        edge_ip_address: Optional[str] = None,
+        edge_user: Optional[str] = None,
+    ):
```

Comment:
Should this be typed to return `RequestOptionBuilder`

Reply:
if I add the return type of self the interpreter complains; it doesn't recognize its own type. They function docs when hovering does recognize the return through

Reply:
https://stackoverflow.com/questions/33533148/how-do-i-type-hint-a-method-with-the-type-of-the-enclosing-class

Reply:
) -> "RequestOptionBuilder":
used the string hint implementation.

Reply:
What version of py do we support for this repo?

Reply:
3.8.
adding an extra import to support not adding quotes seems wrong.

Reply:
Yeah all of it feels a bit weird. Would it be better to just not include it? It seems decently clear without the return type hint.

Reply:
or we can import Self and make that the return. I've only seen quotes added; import Self or future from annotations is odd.

Reply:
I vote for not hinting or just adding the string. for parameters hinting is definitely preferred; returns, I think we just make a call and stick to it.

Reply:
Agreed, remove the hints for this guy